rotkehlchen/chain/evm/decoding/base.py

```diff
@@ -51,24 +52,46 @@ def __init__(
         with self.database.conn.read_ctx() as cursor:
             self.tracked_accounts = self.database.get_blockchain_accounts(cursor)
         self.sequence_counter = 0
+        self.sequence_offset = 0
 
-    def reset_sequence_counter(self) -> None:
+    def reset_sequence_counter(self, tx_receipt: 'EvmTxReceipt') -> None:
+        """Reset the sequence index counter before decoding a transaction.
+        `sequence_offset` is set to one more than the highest tx log index, and is used in
+        `get_next_sequence_index` to add new events whose sequence index will not collide with
+        the sequence index of any events that have associated tx logs.
+        """
         self.sequence_counter = 0
+        self.sequence_offset = tx_receipt.logs[-1].log_index + 1 if len(tx_receipt.logs) else 0
 
-    def get_next_sequence_counter(self) -> int:
-        """Returns current counter and also increases it.
-        Meant to be called for all transaction events that do not have a corresponding log index"""
+    def get_next_sequence_index_pre_decoding(self) -> int:
+        """Get a sequence index for a new event created prior to running the decoding rules.
+        Used for gas events, eth transfers, etc. Must never be used after `get_sequence_index`
+        or `get_next_sequence_index` has been used to prevent sequence index collisions.
+        Returns the current counter and increments it.
+        """
```

Comment:
>         Must never be used after `get_sequence_index`
>        or `get_next_sequence_index` has been used to prevent sequence index collisions.

You can actually check for this in develop time only. Will help us to make sure we are fine for a while. And then we can remove it after a few release cycles. This should not be in production binary as we do `-OO` thanks to pyinstaller's python optimize: https://github.com/rotki/rotki/blob/67c9c0f5166d74611bc6cba65cfeec37708c107b/package.py#L750

```suggestion
        """
        if __debug__:   # develop only test that sequence index was not called
            assert not self.get_sequence_index_called  # Perhaps remove after some time.
```